hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -502,6 +502,8 @@ object HoodieFileIndex extends Logging {
     val isMetadataTableEnabled = getConfigValue(options, sqlConf, HoodieMetadataConfig.ENABLE.key, null)
     if (isMetadataTableEnabled != null) {
       properties.setProperty(HoodieMetadataConfig.ENABLE.key(), String.valueOf(isMetadataTableEnabled))
+    } else {
+      properties.setProperty(HoodieMetadataConfig.ENABLE.key(), String.valueOf(HoodieMetadataConfig.ENABLE.defaultValue()))
```

Comment:
@KnightChess Instead of changing this, should we flip `public static final boolean DEFAULT_METADATA_ENABLE_FOR_READERS = false` to `true` in `HoodieMetadataConfig`?  That controls whether the reader uses metadata table by default.

Reply:
You should also verify if flipping `DEFAULT_METADATA_ENABLE_FOR_READERS` achieves the same.

Reply:
A related PR: #7082

Reply:
> You should also verify if flipping DEFAULT_METADATA_ENABLE_FOR_READERS achieves the same.
make sence, but BigQuery use this param too,  I only considered spark's scene in the begin. Can flipping it if there is no problem with other engines

Reply:
> A related PR: https://github.com/apache/hudi/pull/7082

close this pr.